lib/autoupdate.ps1

```diff
@@ -51,7 +51,7 @@ function find_hash_in_textfile([String] $url, [Hashtable] $substitutions, [Strin
     }
 
     if ($regex.Length -eq 0) {
-        $regex = '^([a-fA-F0-9]+)$'
+        $regex = '^\s*([a-fA-F0-9]+)\s*$'
```

Comment:
Isn't this equivalent to just `([a-fA-F0-9]+)`?

Reply:
I think `^$` exists for some reason.

for example, if the developer releases the checksum with the archive file name, then, the hash will not be catched by this line, but rather by https://github.com/ScoopInstaller/Scoop/blob/master/lib/autoupdate.ps1#L78

This will help avoid errors in some cases, **e.g.** an outdated checksum file.

Reply:
I just tried your example but it doesn't seem to work - https://regex101.com/r/FmSsBb/1

Am I testing it wrong?

Reply:
> I just tried your example but it doesn't seem to work - https://regex101.com/r/FmSsBb/1
> 
> Am I testing it wrong?

Because this is not supposed to be matched by `^\s*([a-fA-F0-9]+)\s*$`, 
but rather by `([a-fA-F0-9]{32,128})[\x20\t]+.*`$basename(?:[\x20\t]+\d+)?` [at line 78](https://github.com/ScoopInstaller/Scoop/blob/master/lib/autoupdate.ps1#L78)

Reply:
Ah okay, I got confused. 

But I have one more confusion. Without the `^$` too, it seems to match correctly (notice the extra whitespace):

![image](https://user-images.githubusercontent.com/46838874/148013238-2cd524d1-b5fb-430f-b47a-6a49fac548d1.png)

Reply:
Oh. I probably didn't explain it clearly enough.

I think the design philosophy of **autoupdate.ps1** is:

(1) Match "*hash*" if the file only contains hash string
       -> this is what I am trying to fix
(2) Convert if the file only contains **Base64** encoded string
(3) Match "filename: *hash*"
      -> This provides an extras check, thus avoiding some problems like an outdated hash file.
(4) Match "hash: *hash*"

This is why we should not use just `([a-fA-F0-9]+)` in step (1).

Reply:
Okay, that makes sense. Thanks for the explanation!